src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -608,4 +609,25 @@ private function assertProductNotDuplicate(Order $order, AddProductToOrderComman
             throw new DuplicateProductInOrderInvoiceException($invoiceNumber, 'You cannot add this product in this invoice as it is already present');
         }
     }
+
+    /**
+     * Checks whether the discounts based on product quantities should be applied when recomputing order details
+     *
+     * @param AddProductToOrderCommand $command
+     *
+     * @return bool
+     */
+    private function shouldQuantityDiscountsBeApplied(AddProductToOrderCommand $command): bool
+    {
+        // PS_QTY_DISCOUNT_ON_COMBINATION === 1 means quantity discount is based on Combinations, otherwise it's based on Products
+        $qtyDiscountOnProducts = (1 !== (int) Configuration::get('PS_QTY_DISCOUNT_ON_COMBINATION'));
```

Comment:
I don't understand why we keep continue using the Configuration class, when we introduce a Configuration adapter to avoid the usage of the legacy class :thinking: 
ping @matks @jolelievre

Reply:
I think it's a common habit we adopted because a lot of adapter classes are useless wrappers and sometimes even lack some of the legacy class capabilities or are worse.

Adapter namespace was a good idea at the beginning but was done badly so we get used to bad adapter classes and simply use them directly in Handlers, because Handlers are inside Adapter.

The initial plan was
1. wrap each legacy class in a Adapter class and use only the Adapter class
2. when modern code rely 100% on Adapter we can `rm -rf` legacy code and re-implement modern code using the adapter interface

But this plan changed, now it's

1. every call to legacy logic must happen in a QueryHandler or CommandHandler
2. when modern code rely 100% on CQRS we can `rm -rf` legacy code and re-implement new handlers that implement the previous handler interface, and swap "old handler" with "new handler"

So the "don't use legacy class, use adapter class instead" rule was built for initial plan and now I don't know if it's still valuable.

Reply:
Maybe but we do have a `PrestaShop\PrestaShop\Core\ConfigurationInterface` interface and it would work in this case So I agree with @PierreRambaud that we should inject the service here

Reply:
We have to use one or the other, but not both. If the Adapter is going to die, we have to deprecate it. 
Otherwise, we have to do the opposite, use the Adapter until the ObjectModel is killed, but only one should be the source of truth ^^'